addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -400,18 +440,45 @@ def create(self, vals_list):
         applicants = super().create(vals_list)
         applicants.sudo().interviewer_ids._create_recruitment_interviewers()
+
+        if (applicants.interviewer_ids.partner_id - self.env.user.partner_id):
+            for applicant in applicants:
+                interviewers_to_notify = applicant.interviewer_ids.partner_id - self.env.user.partner_id
+                notification_subject = _("You have been assigned as an interviewer for %s", applicant.display_name)
+                notification_body = _("You have been assigned as an interviewer for the Applicant %s", applicant.partner_name)
+                applicant.message_subscribe(partner_ids=interviewers_to_notify.ids)
+                applicant.message_notify(
+                    res_id=applicant.id,
+                    model=applicant._name,
+                    partner_ids=interviewers_to_notify.ids,
+                    author_id=self.env.user.partner_id.id,
+                    email_from=self.env.user.email_formatted,
+                    subject=notification_subject,
+                    body=notification_body,
+                    email_layout_xmlid="mail.mail_notification_layout",
+                    record_name=applicant.display_name,
+                    model_description="Applicant",
+                )
+
         # Record creation through calendar, creates the calendar event directly, it will also create the activity.
-        if 'default_activity_date_deadline' in self.env.context:
-            deadline = fields.Datetime.to_datetime(self.env.context.get('default_activity_date_deadline'))
+        if 'default_applicant_calendar_start' in self.env.context:
             for applicant in applicants:
+                defaults = self.env['calendar.event'].with_context(default_applicant_id=applicant.id).default_get(['res_model_id', 'res_id'])
+                activity_vals = {
+                    'res_model_id': defaults['res_model_id'],
+                    'res_id': defaults['res_id'],
+                    'activity_type_id': applicant.activity_type_id.id,
+                }
                 partners = applicant.partner_id | applicant.user_id.partner_id | applicant.department_id.manager_id.user_id.partner_id
                 self.env['calendar.event'].sudo().with_context(default_applicant_id=applicant.id).create({
                     'applicant_id': applicant.id,
                     'partner_ids': [(6, 0, partners.ids)],
                     'user_id': self.env.uid,
-                    'name': applicant.name,
-                    'start': deadline,
-                    'stop': deadline + relativedelta(minutes=30),
+                    'name': applicant.applicant_calendar_title,
+                    'start': applicant.applicant_calendar_start,
+                    'stop': applicant.applicant_calendar_stop,
+                    # 'description': "This should maybe be a field?",
+                    'activity_ids': [(0, 0, activity_vals)],
```

Comment:
put the data in a list and when a finish your loop to call the create method with this list ;)